### ISSUE TITLE
Change in `isa` parsing in `catch`

### ISSUE THREAD

**Issue by yuyichao** [JuliaLang/julia#19987]
I believe this is at least the second time this particular line in the PyCall.jl tests being broken by base parser change....

```jl
julia> :(try catch ex isa(ex, Exception) end)
:(try 
    catch  # REPL[1], line 1:
        ex isa (ex,Exception)
    end)
```
It doesn't seem valid to allow arbitrary binary operator after catch so maybe this can be unbroken?

**Comment by ararslan** [comment]
Note that it works as expected if you do
```julia
:(try catch ex; isa(ex, Exception) end)
```
or
```julia
:(try
  catch ex
      isa(ex, Exception)
  end)
```
But I agree that it seems weird to allow arbitrary binary operators after `catch`.

**Comment by yuyichao** [comment]
Just a reference to the last time this is broken for fun =)

https://github.com/JuliaLang/julia/issues/16517

**Comment by JeffBezanson** [comment]
Fixed by #23487.
